dm/pkg/terror/error_list.go

```diff
@@ -1132,6 +1135,7 @@ var (
 	ErrSyncerUnsupportedStmt                = New(codeSyncerUnsupportedStmt, ClassSyncUnit, ScopeInternal, LevelHigh, "`%s` statement not supported in %s mode", "")
 	ErrSyncerGetEvent                       = New(codeSyncerGetEvent, ClassSyncUnit, ScopeUpstream, LevelHigh, "get binlog event error: %v", "Please check if the binlog file could be parsed by `mysqlbinlog`.")
 	ErrSyncerDownstreamTableNotFound        = New(codeSyncerDownstreamTableNotFound, ClassSyncUnit, ScopeInternal, LevelHigh, "downstream table %s not found", "")
+	ErrSyncerReprocessWithSafeModeFail      = New(codeSyncerReprocessWithSafeModeFail, ClassSyncUnit, ScopeInternal, LevelMedium, "your `safe-mode-duration` in task.yaml is set to 0s, the task can't be re-processed without safe mode currently", "Please stop and re-start this task. If you want to start task successfully, you need set `safe-mode-duration` > `0s`.")
```

Comment:
> you need set `safe-mode-duration` > `0s`

could you explain how should user use `safe-mode-duration = 0s`? In the begining the user can set `safe-mode-duration = 0s`, but when error happens, it must be changed to `safe-mode-duration > 0s`. And the user now have no chance to set it to 0s again, because we don't support "disable safe-mode because beginLocation equal safeModeExitPoint", the task will meet safemode and can't start.

Reply:
As we know, if DM worker arise an abnormal exit and self-start, `Syncer` will enter safe mode.

The user maybe use DM as a transfer tool, and do some operation(sum/count) on the downstream. If DM synced the binlog A(status = 0) to the downstream, and user did some operation(sum/count) and changed its status = 1, then DM worker crashed and re-started, DM will enter safe mode at this moment. This is cause binlog A(status =0) re-sync.

So user can set `safe-mode-duration = 0s` to report an error when DM worker crash and re-start. At this time, they can make some manual adjustments to the business.

Reply:
I mean "safe-mode-duration = 0s" can only be used for once.

Reply:
Oh, I find I forget to add related integration test. I will add it at once!

> So user can set safe-mode-duration = 0s to report an error when DM worker crash and re-start. At this time, they can make some manual adjustments to the business.

Next, user can stop-task, adjust `safe-mode-duration>0s` in `task.yaml`, and start-task successfully.

After that, user can stop-task, adjust `safe-mode-duration=0s` in `task.yaml`, and start-task again. At this time, DM worker exist the `safeModeExitPoint`, user can use it again!

Reply:
The operation is a little complex indeed. But it works and PM suggests.

Reply:
> After that, user can stop-task, adjust safe-mode-duration=0s in task.yaml, and start-task again. At this time, DM worker exist the safeModeExitPoint, user can use it again!

Why DM worker will exit safeModeExitPoint? safeModeExitPoint is always written to checkpoint at its best, so when DM start task, it will enter safe mode. And since user set `safe-mode-duration=0s`, it will report error immediately. (because we don't support "disable safe-mode because beginLocation equal safeModeExitPoint")

Reply:
> DM worker **exist** the safeModeExitPoint, user can use it again!

not

> DM worker will **exit** safeModeExitPoint

I mean DM worker **exist** the safeModeExitPoint, so DM will enter safe mode even if safe-mode-duration=0s(it doesn't report error).

Reply:
This does not behave same as PR description "- set "0s", not step in safe mode. But if dm exists checkpoint, dm must enter safe-mode. **So dm will report an error.**"

Reply:
I am sorry that I didn't illustrate it in the PR description😂: priority `safeModeExitPoint` > `safe-mode-duration` always.

Reply:
> But if dm exists checkpoint, dm must enter safe-mode. So dm will report an error."

I think `exists checkpoint` does not mean `exit safeModeExitPoint`

Reply:
if priority safeModeExitPoint > safe-mode-duration always, why will DM report error the first time it pauses? DM will always have safeModeExitPoint.

Reply:
The latest code reserved the safe-mode logic, but used `compare(exitPoint, beginPosition)` to determine whether need consider `duration=0`, you can see details in https://github.com/pingcap/tiflow/pull/6225/commits/78fcacaf4218f2c083fb445b7ddf53612dcde60f

Reply:
And I withdraw the conclusion that priority `safeModeExitPoint` > `safe-mode-duration`